### PR TITLE
feat: added new component tds-table-input-wrapper

### PR DESCRIPTION
**Pull request by Lunkan89**
## **Describe pull-request**  
Created and started on `tds-table-input-wrapper` component

## **Issue Linking:**  
_Choose one of the following options_
- **Jira:** Add ticket number after `CDEP-`: [CDEP-](https://tegel.atlassian.net/browse/CDEP-)    
- **GitHub:** Include issue link  
- **No issue:** Describe the problem being solved.  

## **How to test**  
_Provide detailed steps for testing, including any necessary setup._
1. Go to...
2. Check in...
3. Run ...

## **Checklist before submission**
- [ ] I have added unit tests for my changes (if applicable)
- [x] All existing tests pass
- [ ] I have updated the documentation (if applicable)

## **Suggested test steps**
- [ ] Browser testing (Chrome, Safari, Firefox) 
- [ ] Keyboard operability
- [ ] Interactive elements have labels.
- [ ] Storybook controls
- [ ] Design/controls/props is aligned with other components 
- [ ] Dark/light mode and variants 
- [ ] Input fields – values should be displayed properly 
- [ ] Events

## **Screenshots**  
_Include before/after screenshots for UI changes._

## **Additional context**  
_Add any other context or feedback requests about the pull-request here._

### CODE REVIEW

File: packages/core/src/components/table/table-input-wrapper/table-input-wrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        }}
+      >
+        {this.renderSlot ? <slot onSlotchange={() => this.handleSlotChange()} /> : null}
+        <tds-icon
+          class="edit-icon"
+          slot="icon"
+          size="16px"
+          name="edit"
+          data-testid="tds-button-testid"
+        ></tds-icon>
```
Comment (theJohnnyMe) [inline packages/core/src/components/table/table-input-wrapper/table-input-wrapper.tsx:54]: As the component will be used for column search too, maybe we should add the "magnifier" icon as a prefix too and then connect it to the some prop where user can define if it is search or edit. The default state would be undefined, so input without any icons. We can speak more about that tomorrow.

File: packages/core/src/components/table/table-input-wrapper/table-input-wrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          'focused-input-wrapper': this.inputFocused,
+        }}
+      >
+        {this.renderSlot ? <slot onSlotchange={() => this.handleSlotChange()} /> : null}
+        <tds-icon
+          class="edit-icon"
+          slot="icon"
+          size="16px"
+          name="edit"
+          data-testid="tds-button-testid"
```
Comment (mJarsater) [inline packages/core/src/components/table/table-input-wrapper/table-input-wrapper.tsx:53]: Should the test-id not come from the user?

File: packages/core/src/components/table/table-input-wrapper/table-input-wrapper.tsx (hunk 1 above)
Comment (theJohnnyMe) [reply packages/core/src/components/table/table-input-wrapper/table-input-wrapper.tsx:54]: As we discussed, we will create a separate component to be used for header cells and rename this one to be used for body cells.

File: packages/core/src/components/table/table-input-wrapper/table-input-wrapper.tsx (hunk 2 above)
Comment (theJohnnyMe) [reply packages/core/src/components/table/table-input-wrapper/table-input-wrapper.tsx:53]: Probably it was just copy/paste thing.

Comment (Lunkan89) [reply packages/core/src/components/table/table-input-wrapper/table-input-wrapper.tsx]: yeah thats on me

**Review by theJohnnyMe — DISMISSED** [review]
Looks OK, lets test it in demo page to see how it behaves ;)

**Comment by theJohnnyMe** [comment]
Warnings on Storybook build, it has to be checked why it happens and are changes done in this PR related to it: 
Screenshot 2024-08-19 at 09 32 16

**Pull request merged**